x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/datastreams/DataLifecycleFeatureSetUsage.java

```diff
@@ -0,0 +1,152 @@
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class DataLifecycleFeatureSetUsage extends XPackFeatureSet.Usage {
+    final LifecycleStats lifecycleStats;
+
+    public DataLifecycleFeatureSetUsage(StreamInput input) throws IOException {
+        super(input);
+        this.lifecycleStats = new LifecycleStats(input);
+    }
+
+    public DataLifecycleFeatureSetUsage(LifecycleStats stats) {
+        super(XPackField.DATA_LIFECYCLE, true, true);
+        this.lifecycleStats = stats;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        lifecycleStats.writeTo(out);
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersion.V_8_500_004;
+    }
+
+    @Override
+    protected void innerXContent(XContentBuilder builder, Params params) throws IOException {
+        super.innerXContent(builder, params);
+        builder.field("count", lifecycleStats.dataStreamsWithLifecyclesCount);
+        builder.field("default_rollover_used", lifecycleStats.defaultRolloverUsed);
+        builder.startObject("retention");
+        builder.field("minimum_millis", lifecycleStats.minRetentionMillis);
+        builder.field("maximum_millis", lifecycleStats.maxRetentionMillis);
+        builder.field("average_millis", lifecycleStats.averageRetentionMillis);
+        builder.endObject();
+    }
+
+    @Override
+    public String toString() {
+        return Strings.toString(this);
+    }
+
+    @Override
+    public int hashCode() {
+        return lifecycleStats.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass() != getClass()) {
+            return false;
+        }
+        DataLifecycleFeatureSetUsage other = (DataLifecycleFeatureSetUsage) obj;
+        return Objects.equals(lifecycleStats, other.lifecycleStats);
+    }
+
+    public static class LifecycleStats implements Writeable {
+        final long dataStreamsWithLifecyclesCount;
+        final long minRetentionMillis;
+        final long maxRetentionMillis;
+        final double averageRetentionMillis;
+        final boolean defaultRolloverUsed;
+
+        public LifecycleStats(
+            long dataStreamsWithLifecyclesCount,
+            long minRetention,
+            long maxRetention,
+            double averageRetention,
+            boolean defaultRolloverUsed
+        ) {
+            this.dataStreamsWithLifecyclesCount = dataStreamsWithLifecyclesCount;
+            this.minRetentionMillis = minRetention;
+            this.maxRetentionMillis = maxRetention;
+            this.averageRetentionMillis = averageRetention;
+            this.defaultRolloverUsed = defaultRolloverUsed;
+        }
+
+        public LifecycleStats(StreamInput in) throws IOException {
+            if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_004)) {
```

Comment:
```suggestion
            if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_006)) {
```

Or maybe we can use minimal supported version here.